ngraph/core/src/op/proposal.cpp

```diff
@@ -140,10 +140,11 @@ void op::v4::Proposal::validate_and_infer_types()
     const auto& class_probs_pshape = get_input_partial_shape(0);
     const auto& class_bbox_deltas_pshape = get_input_partial_shape(1);
     const auto& image_shape_pshape = get_input_partial_shape(2);
-    auto batch_size = class_probs_pshape.to_shape()[0];
+    auto batch_size = class_probs_pshape[0];
     if (class_probs_pshape.is_static() && class_bbox_deltas_pshape.is_static() &&
         image_shape_pshape.is_static())
-        set_output_type(1, get_input_element_type(0), Shape{batch_size * m_attrs.post_nms_topn});
+        set_output_type(
+            1, get_input_element_type(0), PartialShape{batch_size * m_attrs.post_nms_topn});
```

Comment:
Did you replace `Shape` to `PartialShape` because `batch_size` can be dynamic? If so I propose to make similar changes in v0::Proposal, **line 77** of the current file.

Reply:
No, since we have a check at line 144